src/components/Articles/CompanyProvisions.vue

```diff
@@ -0,0 +1,255 @@
+        </v-container>
+      </v-row>
+    </v-container>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Mixins, Watch } from 'vue-property-decorator'
+import { CommonMixin } from '@/mixins'
+
+@Component({})
+export default class CompanyProvisions extends Mixins(CommonMixin) {
+  private draftProvisionsRemoved = false
+  private isEditing = false
+  private haveChanges = false
+  private originalProvisionsRemovedValue = false
+  private isInvalid = false
+  private checkboxColor = this.$vuetify.theme.currentTheme.error
+
+  // Props
+  @Prop({ default: () => { return false } })
+  private provisionsRemoved!: boolean
+
+  // Emitters
+  @Emit('companyProvisionsChanged')
+  private emitCompanyProvisionsChanged (provisionsremoved: boolean): void {}
+
+  @Emit('haveChanges')
+  private emitHaveChanges (haveChanges: boolean): void {}
+
+  // Watchers
+  @Watch('provisionsRemoved')
+  private onProvisionsRemovedPropValueChanged (): void {
+    if (!this.haveChanges) {
+      this.originalProvisionsRemovedValue = this.provisionsRemoved
+      this.draftProvisionsRemoved = this.provisionsRemoved
+      this.emitHaveChanges(this.haveChanges)
+    }
+  }
+
+  @Watch('draftProvisionsRemoved')
+  private onDraftProvisionsRemovedPropValueChanged (): void {
+    this.isInvalid = false
+  }
+
+  private setCompanyProvisionsDone (): void {
+    if (!this.haveChanges && this.draftProvisionsRemoved === this.originalProvisionsRemovedValue) {
+      this.isInvalid = true
+    } else {
+      this.isEditing = false
+      if (this.hasDraftProvisionsRemovedChanged) {
+        this.haveChanges = true
+      } else {
+        this.haveChanges = false
+      }
+      this.emitHaveChanges(this.haveChanges)
+      this.emitCompanyProvisionsChanged(this.draftProvisionsRemoved)
+      this.isInvalid = false
+    }
+  }
+
+  private get hasProvisionsRemovedPropsChanged (): boolean {
+    return this.provisionsRemoved !== this.originalProvisionsRemovedValue
+  }
+
+  private get hasDraftProvisionsRemovedChanged (): boolean {
+    return this.draftProvisionsRemoved !== this.originalProvisionsRemovedValue
+  }
+
+  private cancelCompanyProvisionChange () {
+    this.draftProvisionsRemoved = this.provisionsRemoved
+    this.isInvalid = false
+    this.isEditing = false
+  }
+
+  private resetCompanyProvisions () {
+    this.haveChanges = false
+    this.draftProvisionsRemoved = this.originalProvisionsRemovedValue
+    this.emitCompanyProvisionsChanged(this.draftProvisionsRemoved)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+.action-btns {
+  display: flex;
+  justify-content: flex-end;
+  padding-bottom: 1rem;
+  padding-right: 0.875rem;
+
+  .v-btn + .v-btn {
+    margin-left: 0.5rem;
+  }
+
+  .v-btn {
+    min-width: 6.5rem;
+  }
+
+  .v-btn[disabled] {
+    color: white !important;
+    background-color: #1669bb !important;
```

Comment:
Please use a named colour (CSS variable) for this. See our SCSS theme file.

Reply:
👍